### ISSUE TITLE
Error: Cannot find module './lib/extend'

### ISSUE THREAD

**Issue by bpasero** [microsoft/vscode#114516]
From: https://github.com/microsoft/vscode/runs/1719967310#step:10:1

```
Run yarn npm-run-all --max_old_space_size=4095 -lp compile "electron x64" playwright-install download-builtin-extensions
yarn run v1.22.10
$ D:\a\vscode\vscode\node_modules\.bin\npm-run-all --max_old_space_size=4095 -lp compile "electron x64" playwright-install download-builtin-extensions
[compile                    ] $ node --max_old_space_size=4095 ./node_modules/gulp/bin/gulp.js compile
[electron x64               ] $ node build/lib/electron x64
[playwright-install         ] $ node node_modules/playwright/install.js
[download-builtin-extensions] $ node build/lib/builtInExtensions.js
[download-builtin-extensions] [08:00:04] Syncronizing built-in extensions...
[download-builtin-extensions] [08:00:04] You can manage built-in extensions with the --builtin flag
[download-builtin-extensions] internal/modules/cjs/loader.js:818
[download-builtin-extensions]   throw err;
[download-builtin-extensions]   ^
[download-builtin-extensions] 
[download-builtin-extensions] Error: Cannot find module './lib/extend'
[download-builtin-extensions] Require stack:
[download-builtin-extensions] - D:\a\vscode\vscode\node_modules\node.extend\index.js
[download-builtin-extensions] - D:\a\vscode\vscode\node_modules\gulp-remote-retry-src\index.js
[download-builtin-extensions] - D:\a\vscode\vscode\build\lib\extensions.js
[download-builtin-extensions] - D:\a\vscode\vscode\build\lib\builtInExtensions.js
[download-builtin-extensions]     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:815:15)
[download-builtin-extensions]     at Function.Module._load (internal/modules/cjs/loader.js:667:27)
[download-builtin-extensions]     at Module.require (internal/modules/cjs/loader.js:887:19)
[download-builtin-extensions]     at require (internal/modules/cjs/helpers.js:74:18)
[download-builtin-extensions]     at Object.<anonymous> (D:\a\vscode\vscode\node_modules\node.extend\index.js:3:18)
[download-builtin-extensions]     at Module._compile (internal/modules/cjs/loader.js:999:30)
[download-builtin-extensions]     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
[download-builtin-extensions]     at Module.load (internal/modules/cjs/loader.js:863:32)
[download-builtin-extensions]     at Function.Module._load (internal/modules/cjs/loader.js:708:14)
[download-builtin-extensions]     at Module.require (internal/modules/cjs/loader.js:887:19) {
[download-builtin-extensions]   code: 'MODULE_NOT_FOUND',
[download-builtin-extensions]   requireStack: [
[download-builtin-extensions]     'D:\\a\\vscode\\vscode\\node_modules\\node.extend\\index.js',
[download-builtin-extensions]     'D:\\a\\vscode\\vscode\\node_modules\\gulp-remote-retry-src\\index.js',
[download-builtin-extensions]     'D:\\a\\vscode\\vscode\\build\\lib\\extensions.js',
[download-builtin-extensions]     'D:\\a\\vscode\\vscode\\build\\lib\\builtInExtensions.js'
[download-builtin-extensions]   ]
[download-builtin-extensions] }
[download-builtin-extensions] error Command failed with exit code 1.
[download-builtin-extensions] info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
[compile                    ] [08:00:05] Using gulpfile D:\a\vscode\vscode\gulpfile.js
[compile                    ] [08:00:05] Starting 'compile'...
[compile                    ] [08:00:05] Starting monaco-typecheck ...
```

**Comment by alexdima** [comment]
@joaomoreno This is the GH CI, I can investigate.

**Comment by alexdima** [comment]
I have investigated. Basically, the saved cache `Windows-cacheNodeModules9-3cf1a04d826063786b76a34b212b7967c83750e8` is now poisoned with an invalid archive which is missing this file. The cache was created in https://github.com/microsoft/vscode/runs/1719874631?check_suite_focus=true#step:25:3 
```
Post job cleanup.
C:\windows\System32\tar.exe --posix -z -cf cache.tgz -P -C D:/a/vscode/vscode --files-from manifest.txt
: Can't add archive to itself::debug::File Size: 330335493
Cache saved successfully
```
and every subsequent build which has restored that cache now fails. I have reached out to GH Support about the problem. In the meantime, I have busted the cache. We can wait for them to investigate on their side and depending on that outcome, we can continue using the action as-is or we can investigate creating our own archive like we do in Azure Pipelines.
